parser/spec/response_spec.rb

```diff
@@ -50,4 +50,25 @@
 
     expect { JSONAPI.parse_response!(payload) }.not_to raise_error
   end
+
+  it 'fails when an element is missing type or id' do
+    payload = {
+      'data' => [
+        {
+          'type' => 'articles',
+          'id' => '1',
+          'relationships' => {
+            'author' => {
+              'data' => { 'type' => 'people' }
+            }
+          }
+        }
+      ]
+    }
+
+    expect { JSONAPI.parse_response!(payload) }.to raise_error(
+      JSONAPI::Parser::InvalidDocument,
+      'A resource identifier object MUST contain ["id", "type"] members.'
+    )
+  end
```

Comment:
This test is definitely worth having, but could you add one where the order of `id` and `type` is inverted in the payload, so that the test will fail before this PR and succeed after?